cadc-registry/src/main/java/ca/nrc/cadc/reg/CapabilitiesWriter.java

```diff
@@ -143,12 +146,37 @@ private Element getCapabilityElement(Capability c) {
         
         // extensions
         if (ext) {
-            ret.addContent(c.getExtensionMetadata());
+            for (Element e : c.getExtensionMetadata()) {
+                ret.addContent(deepCopy(e));
+            }
         }
         
         return ret;
     }
     
+    private Attribute deepCopy(Attribute a, Namespace ns) {
+        return new Attribute(a.getName(), a.getValue(), ns);
+    }
+    
+    private Element deepCopy(Element e) {
+        if (!Namespace.NO_NAMESPACE.equals(e.getNamespace())) {
+            throw new UnsupportedOperationException("expected extension metadata to be in " + Namespace.NO_NAMESPACE + ", found: " + e.getNamespace());
+        }
+        
+        Element ret = new Element(e.getName(), Namespace.NO_NAMESPACE);
+        for (Attribute a : e.getAttributes()) {
+            ret.setAttribute(deepCopy(a, Namespace.NO_NAMESPACE));
```

Comment:
A bit odd that a method is used to create a new Attribute here (deepCopy()) but the Element above is just constructed.

Reply:
Attribute and Element have refs to their parent in jdom, hence this complexity.
The new Element (line 166) and new new Attribute (line 158) are the copy part of the deepCopy; the deep part is the recursion (lines 168 and 174). The recursion starts up at line 150.